### PR TITLE
Feature: Peek and Build-Bot Upgrade

### PR DESCRIPTION
**Pull request by Thomas-Boi**
Hello guys,

This is a big upgrade of our build/peek workflow. I thought of splitting it up into multiple PRs but there are a lot of similar aspects (`peek-bot` and `build-bot` has the same superclass, the `footer` upgrade is a part of the `build-bot`). I apologize in advance if this is a lot and feel free to take your time to review this. 

Here is a list of features according to categories:

`peek-bot`:
1. Feature:
	- Now let us know whether an icon contains strokes when uploading them to Icomoon. This is a secondary check to an upcoming checkSvg update
	- Take snapshots of the ICONS version generated by Icomoon. The images seen on the "Generate Font" page of Icomoon is actually what the final font version look like. `peek-bot` now takes a picture of that.
	- Improved messaging to display the images
	- PR title can now be either "new icon" or "update icon". This is merely a semantic change so it looks clearer in the PR tab and release message
2. Things maintainer should know:
	- The glyph counts should match the number of icons selected (see the `peek-bot`'s PR message, I added a list of things we should look out for). If not, something went wrong => could be because color is not stripped properly.
3. Test:
	- See [this PR](https://github.com/Thomas-Boi/devicon/pull/44#issuecomment-894983099) to see a test case where there are NO STROKES in the SVG.
	- See [this PR](https://github.com/Thomas-Boi/devicon/pull/50#issuecomment-894983025) to see a test case where there are STROKES in the SVG.
	
	
`build-bot`:
1. Feature:
	- Now takes picture of the icons generated by Icomoon
	- Also checks for strokes in SVGs and duplicated icons. If these happen, build script will fail.
	- Incorporate `get_release_message` workflow into the `build-bot`. We don't have to run the two scripts separately anymore.
	- `optimizer-bot` (a part of `build-bot`):
		- Will append a `\n` to each SVG after it's optimized (looking at you *nix users)
		- Will prefix the classes -> SVGs can now use classes and `<style>` tags with no issues.
2. Things maintainer should know:
	- The build release message is now included in the PR message. You can copy it by hovering over the top right corner of the release message section.
	- image

	- Please check the files via "Edit Files" to see the `\n` in the SVG files. For some reason, GitHub don't display them but they are there. You can download the file and check locally as well => the `\n` will be there.
	- image
	- image

3. Test:
	- See [this PR](https://github.com/Thomas-Boi/devicon/pull/69) to see the `built-bot`'s result.
	- [Here's](https://github.com/Thomas-Boi/devicon/actions/runs/1125673312) the Actions log for it.

Both the upgrades require merge into `master` due to the `workflow_run` events in the related workflow files.

After this PR, I have another one for `checkSvg`. After that, we can create a new release.

### CODE REVIEW

**Review by amacado — DISMISSED** [review]
This is crazy! 👍🏻 @Thomas-Boi amazing how much effort you bring in!

File: .github/scripts/icomoon_build.py [hunk 1]
```diff
@@ -26,24 +26,29 @@ def main():
...
+        screenshot_folder = filehandler.create_screenshot_folder("./") 
+        runner = BuildSeleniumRunner(args.download_path,
+            args.geckodriver_path, args.headless)
+        runner.build_icons(args.icomoon_json_path, zip_path,
+            icon_svgs, screenshot_folder)
+
         filehandler.extract_files(str(zip_path), args.download_path)
         filehandler.rename_extracted_files(args.download_path)
+
+        print("Creating the release message by querying GitHub API...")
```
Comment (Panquesito7) [inline .github/scripts/icomoon_build.py:49]:
```suggestion
        print("Creating the release message by querying the GitHub API...")
```

File: .github/workflows/post_peek_screenshot.yml [hunk 2]
```diff
@@ -64,18 +80,31 @@ jobs:
...
-            **The maintainers will now take a look at it and decide whether to merge your PR.**
+            The maintainers will now check for:
+            1. The number of Glyphs matches the number of SVGs that were selected. 
+            2. The icons (second group of pictures) look the same as the SVGs (first group of pictures). 
+            3. The icons are of high quality (legible, matches the official logo, etc.)
 
-            Thank you for contributing to Devicon! I hope everything works out and your icons are accepted into the repo.
+            In case of font issues, it might be caused by Icomoon not accepting strokes in the SVGs.  Check this [doc](https://icomoon.io/#faq/importing) for more details and fix the issues as instructed by Icomoon and update this PR once you are done.
+
+            Thank you for contributing to Devicon! I hope that your icons are accepted into the repo.
```
Comment (Panquesito7) [inline .github/workflows/post_peek_screenshot.yml:105]:
```suggestion
            Thank you for contributing to Devicon! I hope that your icons are accepted into the repository.
```

File: .gitignore [hunk 3]
```diff
@@ -4,4 +4,6 @@ node_modules
 geckodriver.log
 __pycache__
 *.pyc
-new_icons.png
\ No newline at end of file
+new_icons.png
+screenshots/
+release_message.txt
```
Comment (Panquesito7) [inline .gitignore:9]:
```suggestion
release_message.txt

```

**Review by Panquesito7 — COMMENTED** [review]
I don't know anything about Python and I'm not that experienced in this area, but overall, everything looks **amazing**! Awesome work, @Thomas-Boi! 😄

File: gulpfile.js [hunk 4]
```diff
@@ -21,7 +23,7 @@ async function createDeviconMinCSS() {
   await createCSSFiles();
 
   let deviconMinPath = path.join(__dirname, finalMinSCSSName);
-  // recall that devicon-alias.scss imported the devicon.css => don't need
+  // recall that devicon-alias.scss imported the devicon-icomoon.css => don't need
```
Comment (Thomas-Boi) [inline gulpfile.js:26]:
```suggestion
  // recall that devicon-alias.scss imported the devicon-base.css => don't need
```

**Comment by Thomas-Boi** [comment]
Two things I forgot to mentioned:
- I renamed the `devicon.css` into `devicon-base.css`. People keep thinking that the `devicon-min.css` is an optimized version of that, which is only half-correct. Hopefully, a rename will make it clearer that this is not the case.
- The images in the `build-bot` test example were reversed. This has been fixed in `e72f344`.

**Review by Panquesito7 — APPROVED** [review]
As I said previously, I don't know anything about Python or related to this area, but the changes mentioned in the first comment look good. Huge thanks and awesome work, @Thomas-Boi! 😄

**Comment by Thomas-Boi** [comment]
Thank you for reviewing the changes! This took me quite a while and I'm happy that everything went well. We can start making a new release and putting the bot to work 😄

**Pull request merged**